src/query/query_parser/query_parser.rs

```diff
@@ -1399,29 +1437,56 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_escaped_field() {
+        let mut schema_builder = Schema::builder();
+        schema_builder.add_text_field(r#"a\.b"#, STRING);
+        let schema = schema_builder.build();
+        let query_parser = QueryParser::new(schema, Vec::new(), TokenizerManager::default());
+        let query = query_parser.parse_query(r#"a\.b:hello"#).unwrap();
+        assert_eq!(
+            format!("{:?}", query),
+            "TermQuery(Term(type=Str, field=0, \"hello\"))"
+        );
+    }
+
     #[test]
     fn test_split_full_path() {
         let mut schema_builder = Schema::builder();
         schema_builder.add_text_field("second", STRING);
         schema_builder.add_text_field("first", STRING);
         schema_builder.add_text_field("first.toto", STRING);
+        schema_builder.add_text_field("first.toto.titi", STRING);
         schema_builder.add_text_field("third.a.b.c", STRING);
         let schema = schema_builder.build();
-        let query_parser = QueryParser::new(schema, Vec::new(), TokenizerManager::default());
+        let query_parser =
+            QueryParser::new(schema.clone(), Vec::new(), TokenizerManager::default());
         assert_eq!(
             query_parser.split_full_path("first.toto"),
-            ("first.toto", "")
+            Some((schema.get_field("first.toto").unwrap(), ""))
         );
         assert_eq!(
-            query_parser.split_full_path("first.titi"),
-            ("first", "titi")
+            query_parser.split_full_path("first.toto.bubu"),
```

Comment:
I don't understand this, split_full_path returns a field_name and json path. 
How can a text field have a json path?

Reply:
split_full_path just splits something into Field and json path. regardless of the field type.

An error is returned if the field is of type text yes has a non empty json path, but that happens downstream.